src/modules/programs/parseValueBySettingType.js

```diff
@@ -9,22 +9,28 @@ import {
 export const parseValueByType = (name, value) => {
     switch (name) {
         case TEI_DOWNLOAD:
-            value = value ? Math.min(maxValues.teiDownload, parseInt(value)) : 0
+            value =
+                isNaN(parseInt(value)) || Math.max(0, parseInt(value)) <= 0
```

Comment:
This is a bit too complicated... `Math.max(0, parseInt(value))` will never be `< 0`, could be cleaned up:

```
Math.min(maxValues.teiDownload, Math.max(0, isNan(value) ? 0 : parseInt(value)))
```

It would also be cleaner to pull this out into a helper function, since it is repeated multiple places.